src/Symfony/Component/DependencyInjection/Compiler/CheckTypeDeclarationsPass.php

```diff
@@ -249,7 +249,11 @@ private function checkType(Definition $checkedDefinition, $value, \ReflectionPar
             return;
         }
 
-        if ('iterable' === $type && (\is_array($value) || is_subclass_of($class, \Traversable::class))) {
+        if ('iterable' === $type && (\is_array($value) || 'iterable' === $class || is_subclass_of($class, \Traversable::class))) {
```

Comment:
the new check would be covered by the suggestion I made in the other comment + this should be replaced by `'array' === $class`, which is missing right now.